File: .sdk/tm/php/utility/ResultHeaders.php
<?php
declare(strict_types=1);

// YahooFinance SDK utility: result_headers

class YahooFinanceResultHeaders
{
    public static function call(YahooFinanceContext $ctx): ?YahooFinanceResult
    {
        $response = $ctx->response;
        $result = $ctx->result;
        if ($result) {
            if ($response && is_array($response->headers)) {
                $result->headers = $response->headers;
            } else {
                $result->headers = [];
            }
        }
        return $result;
    }
}

File: rb/utility/result_headers.rb
# YahooFinance SDK utility: result_headers
module YahooFinanceUtilities
  ResultHeaders = ->(ctx) {
    response = ctx.response
    result = ctx.result
    if result
      if response && response.headers.is_a?(Hash)
        result.headers = response.headers
      else
        result.headers = {}
      end
    end
    result
  }
end
